PatternPal/PatternPal.SyntaxTree/Abstractions/Relation.cs

```diff
@@ -3,13 +3,22 @@
 
 namespace PatternPal.SyntaxTree.Abstractions;
 
+/// <summary>
+/// Represents a relation of type <see cref="_type"/> from <see cref="Source"/> to <see cref="Target"/>.
```

Comment:
I'd rephrase this, as I don't like referencing private members in the public documentation. As a reader of this documentation I'd assume I could access `_type`, which I can't. Perhaps you could say:
```suggestion
/// Represents a relation of type <see cref="RelationType"/> from <see cref="Source"/> to <see cref="Target"/>.
```